File: Sources/CreatorCoreForge/ArchitectureDetector.swift
import Foundation

public enum AppArchitecture {
    case spa
    case mvc
    case mvvm
    case unknown
}

/// Detects simple architecture keywords from a prompt.
public struct ArchitectureDetector {
    public init() {}

    public func detect(in text: String) -> AppArchitecture {
        let lower = text.lowercased()
        if lower.contains("single page") || lower.contains("spa") {
            return .spa
        }
        if lower.contains("model-view-controller") || lower.contains("mvc") {
            return .mvc
        }
        if lower.contains("mvvm") {
            return .mvvm
        }
        return .unknown
    }
}

File: apps/CoreForgeBuild/services/ArchitectureDetector.ts
export type Architecture = 'spa' | 'mvc' | 'mvvm' | 'unknown';

/** Detect simple architecture keywords from a prompt or UI map. */
export class ArchitectureDetector {
  detect(text: string): Architecture {
    const lower = text.toLowerCase();
    if (lower.includes('single page') || lower.includes('spa')) {
      return 'spa';
    }
    if (lower.includes('model-view-controller') || lower.includes('mvc')) {
      return 'mvc';
    }
    if (lower.includes('mvvm')) {
      return 'mvvm';
    }
    return 'unknown';
  }
}
